.github/scripts/trymerge.py

```diff
@@ -988,13 +1052,18 @@ def merge(pr_num: int, repo: GitRepo,
             find_matching_merge_rule(pr, repo)
             pending = pr_get_pending_checks(pr)
             failing = pr_get_failed_checks(pr)
+
             if (not mandatory_only and on_green) and len(failing) > 0:
                 raise RuntimeError(f"{len(failing)} additional jobs have failed, first few of them are: " +
                                    ' ,'.join(f"[{x[0]}]({x[1]})" for x in failing[:5]))
             if (not mandatory_only and on_green) and len(pending) > 0:
                 raise MandatoryChecksMissingError(f"Still waiting for {len(pending)} additional jobs to finish, " +
                                                   f"first few of them are: {' ,'.join(x[0] for x in pending[:5])}")
-            return pr.merge_into(repo, dry_run=dry_run, force=force, comment_id=comment_id)
+            if land_checks:
+                validate_land_time_checks(repo, commit)
+                return pr.merge_into(repo, dry_run=dry_run, force=force, comment_id=comment_id)
+            else:
+                return pr.merge_into(repo, dry_run=dry_run, force=force, comment_id=comment_id)
```

Comment:
Nit (avoids code duplication)
```suggestion
            return pr.merge_into(repo, dry_run=dry_run, force=force, comment_id=comment_id)
```